docs/LemMinX-Extensions.md

```diff
@@ -101,6 +101,39 @@ The [LemMinx Extensions API](https://github.com/eclipse/lemminx/tree/master/org.
 - Formatter with [IFormatterParticipant](https://github.com/eclipse/lemminx/blob/master/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/extensions/format/IFormatterParticipant.java)
 - Symbols with [ISymbolsProviderParticipant](https://github.com/eclipse/lemminx/blob/master/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/extensions/ISymbolsProviderParticipant.java)
 
+## XML Language Server services available for extensions
+XML Language Server extension may need to use standard Language Server features such as commands, documents and ability to manipulate documents. These are available to extensions indirectly via specialized service API.
+
+#### Command Service
+The command service is a wrapper around Language Server client functionality allowing to register/unregister/execute commands via Language Server Protocol.
+Example:
+
+	// Command registration
+	commandService.registerCommand("my-cmd", (params, cancelChecker) -> "executed");
+	
+	// Execute command. Should return "executed"
+	commandService.executeClientCommand(new ExecuteCommandParams("my-cmd", Collections.emptyList())).get()
+	
+	//Unregister command
+	commandService.unregisterCommand("my-cmd");
+	
+The method `executeClientCommand(...)` allows for executing any command on the client which includes commands registered by other Language Servers as well as execute a client command if there is no matching Language Server registered command. For example one could open a Terminal in VSCode with the following:
+	
+	commandService.executeClientCommand(new ExecuteCommandParams("workbench.action.terminal.toggleTerminal", Collections.emptyList()))
+	
+
+See definition of [IXMLCommandService](https://github.com/eclipse/lemminx/blob/master/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/IXMLCommandService.java)
+
+#### Document Provider
+The document provider allows for finding the document from the document URI and listing all XML documents. Note that the documnent provider is only aware of the XML documents it is working on rather than all possible XML documents in a workspace
+
+See definition of [IXMLDocumentProvider](https://github.com/eclipse/lemminx/blob/master/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/IXMLDocumentProvider.java)
+
+#### Validation Service
+The validation service allows for validating documents.
```

Comment:
The validation service allows for triggering validation of all opened XML documents on server side.